lib/memo_wise.rb

```diff
@@ -106,21 +108,55 @@ def reset_memo_wise(method_name, *args, **kwargs)
     if args.empty? && kwargs.empty?
       @_memo_wise.delete(method_name)
     else
-      method = self.class.instance_method(method_name)
-
-      has_arg = MemoWise.has_arg?(method)
-
-      if has_arg && MemoWise.has_kwarg?(method)
-        @_memo_wise[method_name].delete([args, kwargs])
-      elsif has_arg
-        @_memo_wise[method_name].delete(args)
-      else
-        @_memo_wise[method_name].delete(kwargs)
-      end
+      @_memo_wise[method_name].delete(fetch_key(method_name, *args, **kwargs))
     end
   end
 
   def reset_all_memo_wise
     @_memo_wise.clear
   end
+
+  def preset_memo_wise(method_name, *args, **kwargs, &block)
+    validate_memo_wised!(method_name)
+
+    unless block
+      raise ArgumentError,
+            "Pass a block as the value to preset for #{method_name}, #{args}"
+    end
+
+    validate_params!(method_name, args)
+
+    if method(method_name).arity.zero?
+      @_memo_wise[method_name] = block_given? ? yield : nil
```

Comment:
Why do we have these `block_given?` checks if we enforce there's always a block? Or am I misunderstanding what this code is doing?